### PR TITLE
MINOR: remove unused code from MessageTest

### PR DESCRIPTION
**Pull request by chia7712**
noticed this issue when reviewing #9960

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/test/java/org/apache/kafka/common/message/MessageTest.java [hunk 1]
```diff
@@ -1006,12 +977,6 @@ private void verifyWriteSucceeds(short version, Message message) {
         ByteBuffer buf = ByteBuffer.allocate(size * 2);
         ByteBufferAccessor byteBufferAccessor = new ByteBufferAccessor(buf);
         message.write(byteBufferAccessor, cache, version);
-        ByteBuffer alt = buf.duplicate();
-        alt.flip();
-        StringBuilder bld = new StringBuilder();
-        while (alt.hasRemaining()) {
-            bld.append(String.format(" %02x", alt.get()));
-        }
```
Comment (ijuma) [inline clients/src/test/java/org/apache/kafka/common/message/MessageTest.java:1001]: Was this meant to be used in an assert that checks the buffer content?

Comment (chia7712) [reply clients/src/test/java/org/apache/kafka/common/message/MessageTest.java:1001]: It was unused when it was added by https://github.com/apache/kafka/commit/0de61a4683b92bdee803c51211c3277578ab3edf#diff-02a860dc8bbf9772338aaa85b35914b812577d05384eca3af4f5ddb8b5bda4bdR887

```
        message.write(byteBufferAccessor, cache, version);
         ByteBuffer alt = buf.duplicate();
         alt.flip();
         StringBuilder bld = new StringBuilder();
         while (alt.hasRemaining()) {
             bld.append(String.format(" %02x", alt.get()));
         }
         assertEquals("Expected the serialized size to be " + size +
             ", but it was " + buf.position(), size, buf.position());
```

**Review by ijuma — APPROVED** [review]
LGTM, thanks.

**Comment by chia7712** [comment]
```
Build / JDK 15 / org.apache.kafka.clients.consumer.internals.ConsumerNetworkClientTest.blockWhenPollConditionNotSatisfied()
Build / JDK 15 / org.apache.kafka.clients.consumer.internals.FetcherTest.testEarlierOffsetResetArrivesLate()
Build / JDK 15 / org.apache.kafka.connect.mirror.integration.MirrorConnectorsIntegrationSSLTest.testOneWayReplicationWithAutoOffsetSync()
```

unrelated error

**Pull request merged**